### PR TITLE
[WebGPU EP] Batch Norm Implementation

### PR DESCRIPTION
**Pull request by prathikr**
Increases operator coverage for webgpu ep.

### CODE REVIEW

File: onnxruntime/core/providers/webgpu/nn/batch_norm.h [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+class BatchNormalization final : public WebGpuKernel {
+ public:
+  BatchNormalization(const OpKernelInfo& info) : WebGpuKernel(info) {
+    epsilon_ = info.GetAttrOrDefault<float>("epsilon", 1e-5f);
+    momentum_ = info.GetAttrOrDefault<float>("momentum", 0.9f);
+    spatial_ = info.GetAttrOrDefault<int64_t>("spatial", 1);
+    training_mode_ = info.GetAttrOrDefault<int64_t>("training_mode", 0);
+    // NCHW for ai.onnx domain, NHWC for com.ms.internal.nhwc domain
+    format_ = info.GetAttrOrDefault<std::string>("format", "NHWC");
```
Comment (fs-eire) [inline onnxruntime/core/providers/webgpu/nn/batch_norm.h:39]: Does the attribute "format" actually exist?

Comment (guschmue) [reply onnxruntime/core/providers/webgpu/nn/batch_norm.h:39]: Looked at this: ort can rewrite the graph to use NHWC in the kMSInternalNHWCDomain and in that case we'd use this path. It is passed in the template when we register the op.

**Comment by guschmue** [comment]
lgtm. Did you check that the ort unit test is testting the webgpu route?

**Review by guschmue — APPROVED** [review]
I think we should follow Yulong's recommendation of using DataLayout

**Comment by prathikr** [comment]
> lgtm. Did you check that the ort unit test is testting the webgpu route?

yes

**Review by guschmue — APPROVED** [review]
